core/src/price_estimation/orderbook_based.rs

```diff
@@ -26,35 +32,19 @@ impl PriceSource for PricegraphEstimator {
             let pricegraph = Pricegraph::new(orders.iter().map(|order| {
                 order.to_element(account_state.read_balance(order.sell_token, order.account_id))
             }));
-            Ok(self.estimate_prices(tokens, pricegraph))
+            pricegraph.get_prices(tokens).await
         }
         .boxed()
     }
 }
 
-/// Private trait to facilitate testing this module
-#[cfg_attr(test, mockall::automock)]
-trait PriceEstimating {
-    fn estimate_price(&self, pair: TokenPair, volume: f64) -> Option<f64>;
-}
-
-impl PriceEstimating for Pricegraph {
-    fn estimate_price(&self, pair: TokenPair, volume: f64) -> Option<f64> {
-        self.estimate_limit_price(pair, volume)
-    }
-}
-
-impl PricegraphEstimator {
-    pub fn new(orderbook_reader: Arc<dyn StableXOrderBookReading>) -> Self {
-        Self { orderbook_reader }
-    }
-
-    fn estimate_prices(
-        &self,
-        tokens: &[TokenId],
-        pricegraph: impl PriceEstimating,
-    ) -> HashMap<TokenId, u128> {
-        tokens
+use inner::LimitPriceEstimating;
+impl<T: LimitPriceEstimating> PriceSource for T {
```

Comment:
Can't this `impl` block be inside the `mod inner` as well?

Reply:
I think it can but it doesn't have any effect, does it? The reason I put the trait in the module is so that you cannot import it from outside. This is the only purpose of the module so if it doesn't matter for the impl block then it makes more sense to me outside.